Created declaration-block-trailing-semicolon with options

davidtheclark opened the pull request:
Changes the name of the rule, as @jeddy3 mentioned, and adds the option to either expect *or reject* the trailing semicolon -- which I noticed when I re-read that part of the rules gist.

MoOx commented on src/rules/declaration-block-trailing-semicolon/__tests__/index.js:
```diff
@@ -0,0 +1,153 @@
+
+const testTrailingSemicolon = testRule(trailingSemicolon)
+const expectedError = "Expected a trailing semicolon (declaration-block-trailing-semicolon)"
+const rejectedError = "Expected no trailing semicolon (declaration-block-trailing-semicolon)"
```
maybe we can get those messages exported from the rule directly. With es6 we can have multiples values exported: a default one `import rule from "./rule"` and some others like `import rule, {messages} from "./rule"`. That will make maintenance easier. 
In the rule, we can just add `export const messages = { /*...*/ }`

MoOx commented:
You can rebase on master so tests on appveyor pass too.

jeddy3 commented:
Great to see the options go in! I thought about that last night and was going to give it a shot myself today, but you beat me to it :)

MoOx commented:
We should made an "official" recap for rules names. Some guidelines or something. Some simple rules to build rule names.

jeddy3 commented:
I can have a go that, if you like? 

Where would you like it? Just in the readme for now, or a standalone markdown file somewhere?

MoOx commented:
As you want. Maybe we should start in a `docs/naming-rules.md` (+ link to this folder/toc in the README) ?

davidtheclark commented:
I rebased to the 0.1.0 branch, because that had the changes appveyor needed; and I implemented the suggestion you made @MoOx which I thought was a good one.

MoOx commented:
Note that you can do PR from a branch of this repo. Don't bother you to use a fork since you are a owner ;)

davidtheclark commented:
Ok
